### ISSUE TITLE
BinanceWebSocketApiManager.stop_stream() doensn't stop the stream immediately

### ISSUE THREAD

**Issue by ArnoldWolfstein** [oliver-zehentleitner/unicorn-binance-websocket-api#161]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [X] I have searched for other issues with the same problem or similar feature requests. 

#### Select one:
- [X] Bug
- [ ] Feature Request
- [ ] Technical Help
- [ ] Other

#### Environment
- [ ] Are you using the module on a VPS or other Cloud hosting?
- [ ] Are you using the module on a Raspberry Pi?

#### What kind of internet connection do you have?
```
```

#### Average system load (CPU)
```
```

#### Hardware specification 
```
```

#### Operating System? (include version)
- [ ] macOS
- [ ] Windows
- [X] Linux (include flavour)

#### Options
- [X] stream_buffer
- [ ] process_stream_data

#### Which endpoint do you connect?
```
 binance.com
```

#### Python Version Requirement
- [X] I am using Python 3.6.1 or above

#### Exact Python Version?
```
Python 3.8.1
```

#### Pip Version?
```
pip 21.0.1
```

#### UNICORN Binance WebSocket API Version?
```
1.29.0
```

#### Description of your issue

`stop_stream()` doesn't stop the related stream immediately. As we can see the code just stops **restart**. 

https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/blob/c975d49486f93761bbeb01e5a87ff9d4302129f6/unicorn_binance_websocket_api/unicorn_binance_websocket_api_manager.py#L3171

So we need to wait till next `heartbeat/ping-pong`. Maybe this is the design choice but the function name is a little bit confusing then. Maybe we rename it to: `stop_restart_stream()`? Or we can implement  `delete_listen_key_by_stream_id()`  as well as `delete_stream_from_stream_list` in it? Maybe both? Just a suggestion.

**Comment by oliver-zehentleitner** [comment]
No it does not just stops a restart. 
It also activates a stop request with `self.stream_list[stream_id]['stop_request'] = True`

**Comment by ArnoldWolfstein** [comment]
Yes,and when does the stream stop after **activating**?

**Comment by oliver-zehentleitner** [comment]
the stream loop in the socket class (own threads) is checking if there is a new request every round.

as mentioned [here](https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/issues/144#issuecomment-807140494) i want to try a new way to stop streams with `loop.close()`.

**Comment by oliver-zehentleitner** [comment]
implementing `delete_listen_key_by_stream_id()` makes sense i guess!

`delete_stream_from_stream_list()` is not good, because it contains statistics that can help debugging and maybe devs want still have it available.We can implement a frequent_check that cleans the oldest inactive streams if theere are more 1000 entries on the stack.

**Comment by ArnoldWolfstein** [comment]
Yeah, makes sense: `delete_listen_key_by_stream_id()` should be enough.

**Comment by oliver-zehentleitner** [comment]
Just for documentation: https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/issues/137 depends on the same fix.

**Comment by oliver-zehentleitner** [comment]
I dont know whats the right or working way to solve this. Has someone an idea or a hint?
